src/data/polynomial/monic.lean

```diff
@@ -161,16 +161,15 @@ lemma next_coeff_mul {p q : polynomial R} (hp : monic p) (hq : monic q) :
 next_coeff (p * q) = next_coeff p + next_coeff q :=
 begin
   classical,
-  by_cases h : nontrivial R, swap,
-  { rw nontrivial_iff at h, push_neg at h, apply h, },
-  haveI := h, clear h,
-  have := monic.nat_degree_mul hp hq,
-  dsimp only [next_coeff], rw this,
-  simp only [hp, hq, degree_eq_zero_iff_eq_one, add_eq_zero_iff], clear this,
-  split_ifs; try { tauto <|> simp [h_1, h_2] },
-  rename h_1 hp0, rename h_2 hq0, clear h,
-  rw ← degree_eq_zero_iff_eq_one at hp0 hq0, assumption',
-  -- we've reduced to the case where the degrees dp and dq are nonzero
+  obtain _H|_H := subsingleton_or_nontrivial R; resetI,
+  { exact subsingleton.elim _ _ },
```

Comment:
This is what the `nontriviality` tactic is for, isn't it?

Reply:
Indeed, this works:
```suggestion
  nontriviality,
```